ci/builders/README.md

```diff
@@ -297,7 +297,12 @@ default is empty which means no interpreter will be used to run the script
 and it is assumed the script is already an executable with the right
 permissions to run in the target platform.
 * **name** - the name of the step running the script.
-* **parameters** - flags or parameters passed to the script.
+* **parameters** - flags or parameters passed to the script. Parameters
+accept magic environment variables(placeholders replaced before executing
```

Comment:
I am not sure I understand here... So this insert the string literal ${FLUTTER_LOG_DIR} for later interpolation? How do you know then what this variable will be? It has to be set somewhere, why do this instead of letting these be set as passed in?

Reply:
FLUTTER_LOG_DIR is an environment variable set by engine v2 recipes to save logs that will be automatically uploaded to GCS.

Assuming recipes set FLUTTER_LOG_DIR=[tmp]/a/ and we set the parameters to: ['param1', 'param2', '${FLUTTER_LOG_DIR}'] the list of parameters will be translated to ['param1', 'param2', '[tmp]/a'] before the test execution.

This is required because we don't know the sandbox path before the build execution.

Reply:
So this is set internally in the recipe? Is it mandatory?

Reply:
So this is set internally in the recipe?
Yes the env variable is set internally in the recipe. 

Is it mandatory?
No, scripts controlled by flutter should read the env variable directly to know where to put logs. Support for this was added to pass the environment variable value as a script parameter in scripts that we don't own. This is an example: https://github.com/flutter/engine/blob/main/ci/builders/linux_android_aot_engine.json#L114